packages/@aws-cdk/aws-iam/test/test.role.ts

```diff
@@ -309,45 +314,757 @@ export = {
+              {
+                Action: "s3:*",
+                Effect: "Allow",
+                Resource: "xyz",
+              },
+            ],
+          },
+          Roles: ["MyRole"],
+        }));
+
+        test.done();
       },
-      Roles: [ "MyRole" ]
-    }));
-    test.done();
+
+      'attaches policies from a different env-agnostic stack if imported by static ARN with default options'(test: Test) {
+        // GIVEN
+        const roleStack = new Stack();
+        const importedRole = Role.fromRoleArn(roleStack, 'ImportedRole',
+          'arn:aws:iam::123456789012:role/MyRole');
+
+        // WHEN
+        const policyStack = new Stack();
+        importedRole.attachInlinePolicy(new Policy(policyStack, 'MyPolicy', {
+          statements: [new PolicyStatement({
+            actions: ['s3:*'],
+            resources: ['xyz'],
+          })],
+        }));
+
+        // THEN
+        expect(policyStack).to(haveResourceLike('AWS::IAM::Policy', {
+          PolicyDocument: {
+            Statement: [
+              {
+                Action: "s3:*",
+                Effect: "Allow",
+                Resource: "xyz",
+              },
+            ],
+          },
+          Roles: ["MyRole"],
+        }));
+
+        test.done();
+      },
+
+      // tslint:disable-next-line:max-line-length
+      'attaches policies from a different stack if imported by static ARN with default options and the ARN account == policy stack account'(test: Test) {
+        // GIVEN
+        const roleStack = new Stack();
+        const importedRole = Role.fromRoleArn(roleStack, 'ImportedRole',
+          'arn:aws:iam::123456789012:role/MyRole');
+
+        // WHEN
+        const policyStack = new Stack(undefined, undefined, { env: { account: '123456789012' }});
+        importedRole.attachInlinePolicy(new Policy(policyStack, 'MyPolicy', {
+          statements: [new PolicyStatement({
+            actions: ['s3:*'],
+            resources: ['xyz'],
+          })],
+        }));
+
+        // THEN
+        expect(policyStack).to(haveResourceLike('AWS::IAM::Policy', {
+          PolicyDocument: {
+            Statement: [
+              {
+                Action: "s3:*",
+                Effect: "Allow",
+                Resource: "xyz",
+              },
+            ],
+          },
+          Roles: ["MyRole"],
+        }));
+
+        test.done();
+      },
+
+      // tslint:disable-next-line:max-line-length
+      'ignores policies from a different stack if imported by static ARN with default options and the ARN account != policy stack account'(test: Test) {
+        // GIVEN
+        const roleStack = new Stack();
+        const importedRole = Role.fromRoleArn(roleStack, 'ImportedRole',
+          'arn:aws:iam::123456789012:role/MyRole');
+
+        // WHEN
+        const policyStack = new Stack(undefined, undefined, { env: { account: '012345678901' }});
+        importedRole.attachInlinePolicy(new Policy(policyStack, 'MyPolicy', {
```

Comment:
Wow this is a confusing line. Are we testing that Policy does NOT end up with the ARN previously mentioned in the `Roles` key?

I actually think we might want to make `importedRole.attachInlinePolicy` and `new Policy(... { roles: [importedRole] })` equivalent, don't we?